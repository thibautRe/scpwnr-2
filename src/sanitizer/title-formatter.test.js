const formatter = require('./title-formatter')

const fixtures = [
  {
    name: 'Name',
    artist: 'Artist',
  },
  {
    name: 'Name',
    artist: 'Artist',
    remixedBy: 'RemixedBy',
  },
  {
    name: 'Name',
    artist: 'Artist',
    remixedBy: 'RemixedBy',
    ft: 'Featuring',
  },
  {
    artist: 'Artist',
  },
  {
    name: 'Name',
    remixedBy: 'RemixedBy',
  },
  {
    artist: 'artist',
    remixedBy: 'RemixedBy',
  },
]

test('title formatter', () => {
  fixtures.forEach((titleAST) =>
    expect(formatter(titleAST)).toMatchSnapshot(JSON.stringify(titleAST)),
  )
})
