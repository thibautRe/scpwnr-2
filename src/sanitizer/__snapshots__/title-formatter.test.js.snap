// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title formatter: {"artist":"Artist"} 1`] = `"Artist - Untitled"`;

exports[`title formatter: {"artist":"artist","remixedBy":"RemixedBy"} 1`] = `"artist - Untitled (RemixedBy remix)"`;

exports[`title formatter: {"artist":"k?d","name":"Name"} 1`] = `"k?d - Name"`;

exports[`title formatter: {"name":"Name","artist":"Artist","remixedBy":"RemixedBy","ft":"Featuring"} 1`] = `"Artist - Name (RemixedBy remix) (ft. Featuring)"`;

exports[`title formatter: {"name":"Name","artist":"Artist","remixedBy":"RemixedBy"} 1`] = `"Artist - Name (RemixedBy remix)"`;

exports[`title formatter: {"name":"Name","artist":"Artist"} 1`] = `"Artist - Name"`;

exports[`title formatter: {"name":"Name","remixedBy":"RemixedBy"} 1`] = `"Name (RemixedBy remix)"`;
